### ISSUE TITLE
Regression: Cilium drops traffic to VLAN interfaces configured on top of a bond interface

### ISSUE THREAD

**Issue by tongpu** [cilium/cilium#30593]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After we upgraded from Cilium 1.13.2 to 1.14.6 we've been facing a problem on some of our nodes, which are using VLAN interfaces on top of an untagged bond interface:

```mermaid
flowchart TD
    A[bond0\n192.168.18.44/24\nVLAN 36 untagged] --> B[iscsi.2001\n172.16.124.44/24\nVLAN 2001 tagged]
    A --> C[iscsi.2002\n172.16.125.44/24\nVLAN 2002 tagged]
```

Traffic to VLANs 2001 and 2002 was dropped on bond0, as we could see with `hubble observe`:

```
Feb  2 09:17:22.062: 16:70:99:db:d9:69 (unknown) <> ff:ff:ff:ff:ff:ff (unknown) VLAN traffic disallowed by VLAN filter DROPPED (Ethernet)
Feb  2 09:17:22.062: 40:20:9d:07:c2:6e (unknown) <> ff:ff:ff:ff:ff:ff (unknown) VLAN traffic disallowed by VLAN filter DROPPED (Ethernet)
```

After we moved the IP from bond0 to a tagged VLAN interface everything was working as expected again. At no point did we configure `--vlan-bpf-bypass`.

### Cilium Version

1.14.6 4a4fa058 2024-01-17T20:45:58+02:00 go version go1.20.12 linux/amd64

### Kernel Version

5.15.0-92-generic #102~20.04.1-Ubuntu SMP Mon Jan 15 13:09:14 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

1.27.9

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by julianwiedmann** [comment]
Can you upload a dump for the working 1.13 and broken 1.14 case? Have you checked how v1.15 is behaving?

I'm mildy suspicious that in the v1.14 case we now might be attaching bpf_host to the bond interface.

**Comment by tongpu** [comment]
I would have to build an environment to reproduce it, as we already fixed the affected environment.
With dump your talking about sysdump and not PCAP file from tcpdump right?

**Comment by julianwiedmann** [comment]
> I would have to build an environment to reproduce it, as we already fixed the affected environment.

And the "fix" was just to move the IP around - no further cilium config change etc, correct? Then a `cilium status` for the pod on the affected node should already tell you whether bond0 is under cilium control (see whether it's listed under `KubeProxyReplacement: ...`)

> With dump your talking about sysdump and not PCAP file from tcpdump right?

yep a cilium sydump

**Comment by johren-hpe** [comment]
Is there any further information on this issue?  I am hitting the same issue.   With v1.14.9, vlan tagged packets are getting dropped after the bond.   

`Apr  4 21:09:04.313: 0.0.0.0:68 (world) <> 255.255.255.255:67 (world) VLAN traffic disallowed by VLAN filter DROPPED (UDP)`

When I downgrade to v1.13.2, the packets do not get dropped.  

cilium status on v1.14.9

`KubeProxyReplacement:    Strict   [enp166s0f0 172.24.2.7 (Direct Routing)]`

cilium status on v1.13.2

`KubeProxyReplacement:    Strict`

**Comment by johren-hpe** [comment]
Digging deeper, this may not be quite the same issue.   Trying several other 1.14 versions, I found that it is works in all of the other 1.14 versions and the "Direct Routing" interface shown in the cilium status was now the bridge interface (mgmt0).

`KubeProxyReplacement:    Strict   [mgmt0 172.24.2.7 (Direct Routing)]`

I tried 1.14.9 again and this time it also had the bridge interface (mgmt0).   I suspect what happened in this case was that the bond and bridge were not in place when cilium started so it chose the slave interface (enp166s0f0).   For me, it works when it choses the bridge and doesn't work when it choses the slave interface.
